### ISSUE TITLE
render_utils.lua:46: attempt to perform arithmetic on field 'y' 

### ISSUE THREAD

**Issue by cigumo** [cigumo/krli#836]
Somewhere the scale.y value is being set to nil. 
¿could this be the error patched in luajit switch for the recycled table keys? 
```
        sy = sy * f.scale.y


D/CrashlyticsCore Error: ./all/render_utils.lua:46: attempt to perform arithmetic on field 'y' (a nil value) 
stack traceback: 
./all/render_utils.lua:46: in function 'frame_draw_params' 
./all/render_utils.lua:141: in function 'draw_frames_range' 
./all/game.lua:887: in function 'draw_game' 
./all/game.lua:547: in function 'draw' 
./all/director.lua:562: in function 'draw' 
./main.lua:303: in function 'draw' 
./main.lua:442: in function <./main.lua:390> [C]: in function 'xpcall'
```

Updates: 
- happened after 4.2.10 fix, so it's not only in the particles.

**Comment by cigumo** [comment]
http://support.ironhidegames.com/a/tickets/61961
Pixel 4 XL usando Android 10 5 March security patch
resulta que en origins cuando llega a niveles 12-22

![](https://cdn.discordapp.com/attachments/565142610235228172/693078557143990342/Screenshot_20200327-230849.png)

**Comment by cigumo** [comment]
http://support.ironhidegames.com/a/tickets/61961

**Comment by cigumo** [comment]
13a19b1b754ab39d0967b5533c61014415896795 adds a fix for particles that might be related to this. It will be part of 4.2.10

**Comment by cigumo** [comment]
crashed again in 4.2.11, ZenFone Max Pro M2, Android 9
stage 81
```
./all/render_utils.lua:46: attempt to perform arithmetic on field 'y' (a nil value) 
stack traceback: 
./all/render_utils.lua:46: in function 'frame_draw_params' 
./all/render_utils.lua:141: in function 'draw_frames_range' 
./all/game.lua:1015: in function 'draw_game'
./all/game.lua:377: in function 'draw' 
./all/director.lua:612: in function 'draw'
./main.lua:316: in function 'draw' .
/main.lua:516: in function <./main.lua:451> [C]: in function 'xpcall'
```

**Comment by cigumo** [comment]
¿could this be the error patched in luajit switch?

**Comment by cigumo** [comment]
from #1193 : 

the game crashes after 5-10 minutes of playing on survival mode and the user gets a blue screen:

![Screenshot_20200227_093532_com ironhidegames android kingdomrushorigins](https://user-images.githubusercontent.com/49003944/75581479-5812e580-5a48-11ea-8635-3abe2cf23567.jpg)

Huawei p30 lite
Emui 9.1.0

ticket:
http://support.ironhidegames.com/a/tickets/61224

**Comment by cigumo** [comment]
http://support.ironhidegames.com/a/tickets/63494
1. Device Name : xiaomi mi a1
2. OS Version : Android 9, just update may patch
3. Game version : 4.2.11
![Screenshot_20200522-005638](https://user-images.githubusercontent.com/934821/82703552-828ccd00-9c4a-11ea-8332-6d4700ab9bb0.png)

**Comment by cigumo** [comment]
Kalibrus

KR3 Endless mode when a rocket flies into Denas the game crashes

![Screenshot_20200517-163343_Origins](https://user-images.githubusercontent.com/934821/83934598-a4c53580-a788-11ea-8537-825a329c9bd2.jpg)
![Screenshot_20200606-012539_Origins](https://user-images.githubusercontent.com/934821/83934603-aabb1680-a788-11ea-9422-0b7cc4e8ac29.jpg)

**Comment by cigumo** [comment]
fixed in 4.2.19
